frontend/lib/src/dataframes/arrowFormatUtils.ts

```diff
@@ -144,89 +151,147 @@ interface Interval {
+  timestamp: number | bigint,
   unit: number
 ): number {
   let unitAdjustment
 
-  if (unit === 1) {
+  if (unit === TimeUnit.MILLISECOND) {
     // Milliseconds
     unitAdjustment = 1000
-  } else if (unit === 2) {
+  } else if (unit === TimeUnit.MICROSECOND) {
     // Microseconds
     unitAdjustment = 1000 * 1000
-  } else if (unit === 3) {
+  } else if (unit === TimeUnit.NANOSECOND) {
     // Nanoseconds
     unitAdjustment = 1000 * 1000 * 1000
   } else {
     // Interpret it as seconds as a fallback
-    return Number(value)
+    return Number(timestamp)
   }
 
   // Do the calculation based on bigints, if the value
   // is a bigint and not safe for usage as number.
   // This might lose some precision since it doesn't keep
   // fractional parts.
-  if (typeof value === "bigint" && !Number.isSafeInteger(Number(value))) {
-    return Number(value / BigInt(unitAdjustment))
+  if (
+    typeof timestamp === "bigint" &&
+    !Number.isSafeInteger(Number(timestamp))
+  ) {
+    return Number(timestamp / BigInt(unitAdjustment))
   }
 
-  return Number(value) / unitAdjustment
+  return Number(timestamp) / unitAdjustment
 }
 
-function formatTime(data: number, field?: Field): string {
-  const timeInSeconds = convertToSeconds(data, field?.type?.unit ?? 0)
-  return moment
-    .unix(timeInSeconds)
+/**
+ * Converts a timestamp to a date object.
+ *
+ * @param timestamp The timestamp to convert.
+ * @param field The field containing the unit information.
+ * @returns The date object.
+ */
+export function convertTimestampToDate(
+  timestamp: number | bigint,
+  field?: Field
+): Date {
+  // Time values from arrow are not converted to a shared unit and
+  // just return the raw arrow value. Therefore, we need to adjust
+  // the value to seconds based on the unit information in the field.
+  // https://github.com/apache/arrow/blob/9e08c57c0986531879aadf7942998d26a94a5d1b/js/src/visitor/get.ts#L193C7-L209
+  const timeInSeconds = convertTimestampToSeconds(
+    timestamp,
+    field?.type?.unit ?? TimeUnit.SECOND
+  )
+  return moment.unix(timeInSeconds).utc().toDate()
+}
+
+/**
+ * Formats a time value based on the unit information in the field.
+ *
+ * @param timestamp The time value to format.
+ * @param field The field containing the unit information.
+ * @returns The formatted time value.
+ */
+function formatTime(timestamp: number | bigint, field?: Field): string {
+  const date = convertTimestampToDate(timestamp, field)
+  return moment(date)
     .utc()
-    .format(timeInSeconds % 1 === 0 ? "HH:mm:ss" : "HH:mm:ss.SSS")
+    .format(date.getMilliseconds() === 0 ? "HH:mm:ss" : "HH:mm:ss.SSS")
 }
 
-function formatDuration(data: number | bigint, field?: Field): string {
+function formatDate(date: number | Date): string {
+  const formatPattern = "YYYY-MM-DD"
+
+  // Date values from arrow are already converted to a date object
+  // or a timestamp in milliseconds even if the field unit might indicate a
+  // different unit.
```

Comment:
nit: it looks like `"even if the field unit might indicate a different unit."` is a copy-paste artifact from the `formatDatetime` function because here we don't have a the `unit` argument

Reply:
theoretically, we also have the unit for this field. It just doesn't get passed to `formatDate` since we don't need it for formatting of date values. And it's not needed because arrow already takes care of unifying the date values to date or milliseconds timestamp. But the big issue here: Arrow is not super consistent on this part, some field types are unified but other required to actually do it on our side by using the field unit. So its a bit of a case-by-case basis by looking into the actual ArrowJS implementation.

Reply:
thanks for the context! However, the field is not even passed to this function, so at this place we don't have it.

Reply:
I added an additional comment. I think it's still important to keep the information in this function why it doesn't need the field information at all.

Reply:
Yeah I think it's great context! And thanks for updating the comment 👍 I would do a small modification in the sense of:  

```javascript
// ... even if the field unit belonging to the passed date might have indicated a different unit.
// Thats why we don't need the field information here (aka its not passed to the function)
// and we don't need to apply any unit conversion.
// https://github.com/apache/arrow/blob/9e08c57c0986531879aadf7942998d26a94a5d1b/js/src/visitor/get.ts#L167-L171
```

Reply:
Changed 👍